src/main/java/io/github/cottonmc/cotton/gui/widget/WTextField.java

```diff
@@ -34,9 +34,78 @@
 	protected String text = "";
 	protected int maxLength = 16;
 	protected boolean editable = true;
+	protected boolean drawFocusBorder = true;
+	protected boolean drawTextWithShadow = true;
 	
 	protected int enabledColor = 0xE0E0E0;
 	protected int uneditableColor = 0x707070;
+	protected int backgroundColor = 0xFF_000000;
+	protected int focusedColor = 0xFF_FFFFA0;
+	private boolean resize;
+	
+	public boolean isDrawFocusBorder() {
+		return drawFocusBorder;
+	}
+	
+	public void setDrawFocusBorder(boolean drawFocusBorder) {
+		this.drawFocusBorder = drawFocusBorder;
+	}
+	
+	public boolean isDrawTextWithShadow() {
+		return drawTextWithShadow;
+	}
+	
+	public void setDrawTextWithShadow(boolean drawTextWithShadow) {
+		this.drawTextWithShadow = drawTextWithShadow;
+	}
+	
+	public int getSuggestionColor() {
+		return suggestionColor;
+	}
+	
+	public void setSuggestionColor(int suggestionColor) {
+		this.suggestionColor = suggestionColor;
+	}
+	
+	protected int suggestionColor = -8355712;
+	
+	public int getEnabledColor() {
+		return enabledColor;
+	}
+	
+	public int getUneditableColor() {
+		return uneditableColor;
+	}
+	
+	public void setUneditableColor(int uneditableColor) {
+		this.uneditableColor = uneditableColor;
+	}
+	
+	public int getBackgroundColor() {
+		return backgroundColor;
+	}
+	
+	public void setBackgroundColor(int backgroundColor) {
+		this.backgroundColor = backgroundColor;
+	}
+	
+	public int getFocusedColor() {
+		return focusedColor;
+	}
+	
+	public void setFocusedColor(int focusedColor) {
+		this.focusedColor = focusedColor;
+	}
+	
+	public int getNotFocusedColor() {
+		return notFocusedColor;
+	}
+	
+	public void setNotFocusedColor(int notFocusedColor) {
+		this.notFocusedColor = notFocusedColor;
+	}
```

Comment:
All new methods here should also be moved to the end of the class. Setters should return this `WTextField` instance for builder-like syntax.